org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/SearchableEnvironment.java

```diff
@@ -102,10 +104,10 @@ public SearchableEnvironment(JavaProject project, org.eclipse.jdt.core.ICompilat
 			|| !JavaCore.IGNORE.equals(project.getOption(JavaCore.COMPILER_PB_DISCOURAGED_REFERENCE, true));
 		this.workingCopies = workingCopies;
 		this.nameLookup = project.newNameLookup(workingCopies, excludeTestCode);
-		if (CompilerOptions.versionToJdkLevel(project.getOption(JavaCore.COMPILER_COMPLIANCE, true)) >= ClassFileConstants.JDK9) {
```

Comment:
I've removed that continuous checking of same condition and merged three if blocks in one.
Boy Scout rule. It was dirty before, we should not add more dirt. It is cleaner now.